lib/kafka/client.rb

```diff
@@ -51,6 +52,16 @@ def get_producer(**options)
       Producer.new(cluster: @cluster, logger: @logger, **options)
     end
 
+    def get_async_producer(delivery_interval: 0, delivery_threshold: 0, **options)
```

Comment:
also this method should validate that delivery_interval/threshold are numbers and not negatives, right? (and exception if they are bad). Otherwise things could get weird.